pkg/config/config.go

```diff
@@ -4,17 +4,36 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"reflect"
 	"strings"
 
-	"github.com/bacalhau-project/bacalhau/pkg/config/types"
 	"github.com/mitchellh/mapstructure"
 	"github.com/spf13/viper"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/configenv"
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
 )
 
+var _ Context = (*config)(nil)
+
+type Context interface {
+	RepoPath() (string, error)
+	Load(path string) error
+	Current() (types.BacalhauConfig, error)
+
+	Set(key string, value interface{})
+	SetIfAbsent(key string, value interface{})
+	ForKey(key string, cfg interface{}) error
+
+	User() *viper.Viper
+	System() *viper.Viper
+}
```

Comment:
I am not sure what value this interface is providing. It is tightly coupled to viper and strictly file based config. Also how should we access config? Through `Current()` or `User()`?